### PR TITLE
[Merged by Bors] - fix: backward compatibility for `SmartModuleKind`

### LINKED ISSUE #3035: regression: SmartModule transform
Smartmodule transform doesn't work with `0.10.4` release. 

Steps to reproduce:
1.  Build/download 0.10.5(dev) version of Fluvio
2. Connect to 0.10.4 to Fluvio Cluster
3. Try run Jolt transforms with consume. ```fluvio consume helsinki --transform jolt```

In the 0.10.4 client to 0.10.4 cluster, everything works.   However, connecting 0.10.5 to 0.10.4 doesn't work with transforms.

### PR DESCRIPTION
**Pull request by galibey**
Fixes #3035

### CODE REVIEW

File: crates/fluvio-spu-schema/src/server/smartmodule.rs [hunk 1]
```diff
@@ -94,6 +94,15 @@ pub enum SmartModuleKind {
     },
     #[fluvio(min_version = ARRAY_MAP_WASM_API)]
     FilterMap,
+    #[fluvio(min_version = SMART_MODULE_API)]
```
Comment (morenol) [inline crates/fluvio-spu-schema/src/server/smartmodule.rs:97]: i think that we can put min_version and max_version in the same line like in https://github.com/infinyon/fluvio/blob/ed17bb1532cb493cd14d044d16ac18b15b9c4072/crates/fluvio-protocol/tests/version.rs#L8

Comment (galibey) [reply crates/fluvio-spu-schema/src/server/smartmodule.rs]: You are right. Thanks. Fixed.

**Comment by morenol** [comment]
Not a task for this PR. But maybe we should add max_version here too https://github.com/infinyon/fluvio/blob/ed17bb1532cb493cd14d044d16ac18b15b9c4072/crates/fluvio-spu-schema/src/server/stream_fetch.rs#L74

I see that there are still references to DerivedStreams on CLI

**Comment by sehz** [comment]
confirm fix works.

**Review by sehz — APPROVED** [review]
Fix works against 0.10.4

**Pull request closed without merging**